meteor/sofie-server/data-access/repositories/test/mongo-test-database.ts

```diff
@@ -0,0 +1,91 @@
+// eslint-disable-next-line node/no-unpublished-import
+import { MongoMemoryServer } from 'mongodb-memory-server'
+import { Db, MongoClient } from 'mongodb'
+import { Rundown } from '../../../model/entities/rundown'
+import { MongoEntityConverter } from '../mongo/mongo-entity-converter'
+import { Segment } from '../../../model/entities/segment'
+import { Part } from '../../../model/entities/part'
+import { Piece } from '../../../model/entities/piece'
+import { MongoDatabase } from '../mongo/mongo-database'
+import { when } from 'ts-mockito'
+
+export class MongoTestDatabase {
+	private mongoServer: MongoMemoryServer
+	private client: MongoClient
+
+	constructor() {
+		// Set a timeout beyond the default of 5 Seconds to ensure CI tests don't exceed the limit on GitHub
+		jest.setTimeout(15000)
+	}
+
+	public async setupDatabase(override?: () => Promise<void>) {
+		if (override) {
+			return override()
+		}
+		this.mongoServer = await MongoMemoryServer.create()
+		this.client = await MongoClient.connect(this.mongoServer.getUri())
+	}
+
+	public async teardownDatabase(override?: () => Promise<void>) {
+		if (override) {
+			return override()
+		}
+		if (this.client) {
+			await this.client.close()
+		}
+		if (this.mongoServer) {
+			await this.mongoServer.stop()
+		}
+	}
+
+	public getDatabase(): Db {
+		return this.client.db(this.mongoServer.instanceInfo!.dbName)
```

Comment:
Why is instanceInfo non-null asserted here? To me we are not sure that that is the case.

Reply:
That question i'd like a hand over to @RasmusAlbrektsen, as he was the one to add that originally, and propably had some reason for it.

That said, i haven't personally checked what happens if the exclamation mark is removed.

Reply:
Just checked what happens without the exclamation mark. 
The `instanceInfo` is of type `MongoInstanceData | undefined`.